pipeline_manager/src/db.rs

```diff
@@ -222,59 +257,55 @@ impl ProjectDB {
     }
 
     /// Retrieve code of the specified project.
-    pub(crate) async fn project_code(&self, project_id: ProjectId) -> AnyResult<(Version, String)> {
-        let row = self
+    pub(crate) fn project_code(&self, project_id: ProjectId) -> AnyResult<(Version, String)> {
+        let (version, code) = self
             .dbclient
-            .query_opt(
-                "SELECT version, code FROM project WHERE id = $1",
-                &[&project_id.0],
+            .query_row(
+                "SELECT version, code FROM project WHERE id = ?",
+                [&project_id.0],
+                |row| Ok((Version(row.get(0)?), row.get(1)?)),
             )
-            .await?
-            .ok_or(DBError::UnknownProject(project_id))?;
+            .map_err(|_| DBError::UnknownProject(project_id))?;
 
-        Ok((Version(row.try_get(0)?), row.try_get(1)?))
+        Ok((version, code))
     }
 
     /// Create a new project.
-    pub(crate) async fn new_project(
+    pub(crate) fn new_project(
         &self,
         project_name: &str,
         project_code: &str,
     ) -> AnyResult<(ProjectId, Version)> {
-        let row = self
-            .dbclient
-            .query_one("SELECT nextval('project_id_seq')", &[])
-            .await?;
-        let id: ProjectId = ProjectId(row.try_get(0)?);
-
         self.dbclient
             .execute(
-                "INSERT INTO project (id, version, name, code, status_since) VALUES($1, 1, $2, $3, now())",
-                &[&id.0, &project_name, &project_code],
-            )
-            .await?;
+                "INSERT INTO project (version, name, code, status_since) VALUES(1, ?, ?, unixepoch('now'))",
```

Comment:
I didn't know you can use ?. What does it mean?

Reply:
Parameterized statement that takes values from the tuple in the next line

Reply:
I should use unified syntax for all queries though.  This got messed up in the PG -> SQlite transition